lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -2078,6 +2078,17 @@ static bool foldCommonMuxValue(MuxOp op, bool isTrueOperand,
       return false;
     }
 
+    auto isARecursiveMux = [](Value v) {
+      if (v.getDefiningOp())
+        if (auto muxOp = dyn_cast<MuxOp>(v.getDefiningOp()))
+          return muxOp.getTrueValue() == v || muxOp.getFalseValue() == v;
```

Comment:
Nit: You can combine these into a single call:

```suggestion
      if (auto muxOp = v.getDefiningOp<MuxOp>())
        return muxOp.getTrueValue() == v || muxOp.getFalseValue() == v;
```

Reply:
Oh cool, thanks!